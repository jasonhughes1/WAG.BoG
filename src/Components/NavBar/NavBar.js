import React from 'react';
import { NavLink } from 'react-router-dom';
import './NavBar.css'


const NavBar = () => {
  return(
    <div className='nav-bar'>
      <h1>WAG.BoG</h1>
      <NavLink className='nav' to='/'>Home</NavLink>
      <NavLink className='nav' to='/search'>Search</NavLink>
      <NavLink className='nav' to='/map'>Map</NavLink>
    </div>
  )
}


export default NavBar;
