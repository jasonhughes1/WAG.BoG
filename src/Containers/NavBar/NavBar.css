.nav-bar {
  display: flex;
  justify-content: space-around;
  width: 100%;
  background-color: #273165;
  padding-top: 30px;
  height: 60px; }

.nav {
  color: white; }

.active {
  color: red; }

.main-header {
  color: white; }
