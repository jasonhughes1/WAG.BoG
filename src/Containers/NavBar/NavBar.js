import React,  { Component } from 'react';
import { NavLink } from 'react-router-dom';
import './NavBar.css';
import { connect } from 'react-redux';
import { postLocation } from '../../Helpers/fetchHelper';
import { getDogs, searchCurrentDogs } from '../../Actions/Actions';
import PropTypes from 'prop-types';

export class NavBar extends Component {


  componentDidMount() {
    this.props.getAllDogs();
    this.props.getCurrentLocation();
  }

  render() {
    return (
      <div className='nav-bar'>
        <h1 className='main-header'>WAG.BoG</h1>
        <NavLink className='nav' exact to='/'>Home</NavLink>
        <NavLink className='nav' to='/search'>Search</NavLink>
        <NavLink className='nav' to='/map'>Map</NavLink>
      </div>
    );
  }
}

export const mapStateToProps = (store) => {
  return {
    dogs: store.dogs,
    location: store.location
  };
};

export const mapDispatchToProps = (dispatch) => {
  return {
    getAllDogs: (newDogs) => {
      dispatch(getDogs(newDogs));
    },
    getCurrentLocation: (currLocation) => {
      dispatch(postLocation(currLocation));
    },
    getSearchDogs: (newDogs) => {
      dispatch(searchCurrentDogs(newDogs));
    }
  };
};

NavBar.propTypes = {
  dogs: PropTypes.array,
  getAllDogs: PropTypes.func,
  getCurrentLocation: PropTypes.func,
  getSearchDogs: PropTypes.func,
  history: PropTypes.object,
  location: PropTypes.array,
  match: PropTypes.object
};


export default connect(mapStateToProps, mapDispatchToProps)(NavBar);
