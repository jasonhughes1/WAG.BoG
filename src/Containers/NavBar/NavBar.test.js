import React from 'react';
import { shallow } from 'enzyme';
import { NavBar, mapStateToProps, mapDispatchToProps } from './NavBar';

describe('NavBar tests', () => {
  let navbar;
  beforeEach( () => {
    const mockFunc = jest.fn();
    navbar = shallow(
      <NavBar
        dogs={['pups']}
        getAllDogs={mockFunc}
        getCurrentLocation={mockFunc}
        history={{key: 'value'}}
        location={['location']}
        match={{key: 'value'}}
      />);
  });

  it('should match the snapshot', () => {
    expect(navbar).toMatchSnapshot();
  });

  it('Should be defined', () => {
    expect(navbar).toBeDefined();
  });
});

describe('map state to props', () => {
  it('should receive dogs from the store', () => {
    const mockStore = {
      dogs: [{name: 'Brooklyn'}]
    };
    const expected = mapStateToProps(mockStore);
    expect(expected.dogs).toEqual(mockStore.dogs);
  });

  it('should receive location from the store', () => {
    const mockStore = {
      location: []
    };
    const expected = mapStateToProps(mockStore);
    expect(expected.location).toEqual(mockStore.location);
  });
});

describe('mapDispatchToProps', () => {
  let mockDispatch;
  let result;

  beforeEach(() => {
    mockDispatch = jest.fn();
    result = mapDispatchToProps(mockDispatch);
  });

  it('should call dispatch when getAllDogs is called', () => {
    result.getAllDogs();
    expect(mockDispatch).toHaveBeenCalled();
  });

  it('should call dispatch when getCurrentLocation is called', () => {
    result.getCurrentLocation();
    expect(mockDispatch).toHaveBeenCalled();
  });

  it('should call dispatch when getSearchDogs is called', () => {
    result.getSearchDogs();
    expect(mockDispatch).toHaveBeenCalled();
  });
});
