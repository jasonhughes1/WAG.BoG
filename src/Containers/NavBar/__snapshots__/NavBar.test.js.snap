// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    dogs={
        Array [
            "pups",
          ]
    }
    getAllDogs={[Function]}
    getCurrentLocation={[Function]}
    history={
        Object {
            "key": "value",
          }
    }
    location={
        Array [
            "location",
          ]
    }
    match={
        Object {
            "key": "value",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          WAG.BoG
</h1>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav"
          exact={true}
          to="/"
>
          Home
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav"
          to="/search"
>
          Search
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav"
          to="/map"
>
          Map
</NavLink>,
      ],
      "className": "nav-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "WAG.BoG",
        },
        "ref": null,
        "rendered": "WAG.BoG",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Home",
          "className": "nav",
          "exact": true,
          "to": "/",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Search",
          "className": "nav",
          "to": "/search",
        },
        "ref": null,
        "rendered": "Search",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Map",
          "className": "nav",
          "to": "/map",
        },
        "ref": null,
        "rendered": "Map",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            WAG.BoG
</h1>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav"
            exact={true}
            to="/"
>
            Home
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav"
            to="/search"
>
            Search
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav"
            to="/map"
>
            Map
</NavLink>,
        ],
        "className": "nav-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "WAG.BoG",
          },
          "ref": null,
          "rendered": "WAG.BoG",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Home",
            "className": "nav",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Search",
            "className": "nav",
            "to": "/search",
          },
          "ref": null,
          "rendered": "Search",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Map",
            "className": "nav",
            "to": "/map",
          },
          "ref": null,
          "rendered": "Map",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
